src/validators/stop_times.rs

```diff
@@ -30,11 +38,40 @@ pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
         })
     });
 
-    wrong_stops.into_iter().map(|(_, v)| v).collect()
+    wrong_stops.into_iter().map(|(_, v)| v)
+}
+
+// All stop_sequence of the stop times of a given trip should be different
+fn check_valid_stop_sequence(gtfs: &gtfs_structures::Gtfs) -> impl Iterator<Item = Issue> + '_ {
+    gtfs.trips.values().filter_map(|trip| {
+        let stop_with_same_sequence = trip
+            .stop_times
+            .iter()
+            .tuple_windows()
+            .filter_map(|(s1, s2)| {
```

Comment:
If we have `0` in all sequence, we’ll have a lot of identical. But that’s probably excessive optimisation (or maybe https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.group_by and filter by len > 1 ?)

However a comment explaining that it works because the data is sorted by sequence would be nice

Reply:
oups you're right, I was planning to add it, but forgot :facepalm: 

I was thinking about the optimisation, but I'm not sure it's worth it, it's quite a big bug to have wrong stop_sequence, I think it should be quite lmited.